### DISCUSSION TITLE
Code refactoring

### DISCUSSION THREAD

**Discussion by ik9999** [highfestiva/finplot#280] [General]
Right now finplot library is basically one file with 2792 lines. It can be confusing to work with such a large file of source code.
I refactored it a little: 
https://github.com/ik9999/finplot/tree/master/finplot

Moved all the classes to [the separate files](https://github.com/ik9999/finplot/tree/master/finplot/lib/classes).
And internal functions to a [separate module](https://github.com/ik9999/finplot/blob/master/finplot/lib/functions/__init__.py).

I can make a pull request if its ok.

Best regards.

**Comment by highfestiva** [comment]
First of all: thanks for your desire to help! But. It's easier for me to maintain a single file rather than 20 smaller ones. I'm the sole maintainer, and that feels a bit fragile. If you're willing to step in and help maintain it, we could go this way in the future.

Since you've looked through the code, you see that I focus on getting things done with a minimum of comments while keeping it somewhat generic. The internal data structure is something of a mess, but it works.

I also want proper commit messages, according to the git standard. Things like "fix" doesn't cut it.

There's no loose coupling between the classes/functions, so such a change would by necessity almost always require edits in two or more files for a single fix/change. Object-orientation is often bad, and as a way to organize code is not ideal. The reason stuff is hard coupled sometimes is not because I want to.

Sorry to put a damper on your enthusiasm, but realistically putting stuff into two-three different files might make it easier, but I think that 20 files would just make it many times harder to refactor and fix. What do you think?
